File: CPP/Program/CPU.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class CPU : public Komponen{
	private:
		int jumlah_core;
		float kecepatan_GHz;
		
	public:
		CPU(){
			
		}
		
		CPU(int jumlah_core, float kecepatan_GHz, string merk, string nama) : 
		Komponen(merk, nama){
			this->jumlah_core = jumlah_core;
			this->kecepatan_GHz = kecepatan_GHz;
		}
		
		void setJumlahCore(int jumlah_core){
			this->jumlah_core = jumlah_core;
		}
		
		void setKecepatanGHz(float kecepatan_GHz){
			this->kecepatan_GHz = kecepatan_GHz;
		}
		
		int getJumlahCore(){
			return this->jumlah_core;
		}
		
		float getKecepatanGHz(){
			return this->kecepatan_GHz;
		}
		
		string display(){
			return getMerk() + ' ' + getNama() + " (" + to_string(getJumlahCore()) + " Core) ~" + to_string(getKecepatanGHz()) + "GHz";
		}
		
		~CPU(){
			
		}
};

File: Python/CPU.py
from Komponen import Komponen

class CPU(Komponen):
    def __init__(self, jumlah_core=0, kecepatan_GHz=0.0, merk="", nama=""):
        super().__init__(merk, nama)
        self._jumlah_core = jumlah_core
        self._kecepatan_GHz = kecepatan_GHz

    def set_jumlah_core(self, jumlah_core):
        self._jumlah_core = jumlah_core

    def set_kecepatan_GHz(self, kecepatan_GHz):
        self._kecepatan_GHz = kecepatan_GHz

    def get_jumlah_core(self):
        return self._jumlah_core

    def get_kecepatan_GHz(self):
        return self._kecepatan_GHz

    def display(self):
        return f"{self.get_merk()} {self.get_nama()} ({self.get_jumlah_core()} Core) ~ {self.get_kecepatan_GHz()} GHz"